src/test/task.test.js

```diff
@@ -0,0 +1,41 @@
+import { addTask, removeTask } from '../task.js';
+
+describe('add a task method', () => {
+  // Arrange
+  const tasks = [];
+  const description = 'the first task';
+  const output = { description, index: 1, completed: false };
+
+  // Act
+  addTask(tasks, description);
+  // Assert
+  test('should add a task', () => {
+    expect(tasks.length).toBe(1);
+  });
+
+  test('should add the correct object', () => {
+    expect(tasks[0]).toMatchObject(output);
+  });
+});
+
+describe('Remove a task method', () => {
+  // Arrange
+  let tasks = [
+    { description: 'New Task', index: 1, completed: false },
+    { description: 'Second Task', index: 2, completed: true },
+    { description: 'Third Task', index: 3, completed: false }];
+
+  const output = [{ description: 'New Task', index: 1, completed: false },
+    { description: 'Third Task', index: 3, completed: false }];
+
+  // Act
+  tasks = removeTask(tasks, 2);
+  // Assert
+  test('Delete an item', () => {
+    expect(tasks.length).toBe(2);
+  });
+  test('Delete the exact item', () => {
+    expect(tasks[0]).toMatchObject(output[0]);
+    expect(tasks[1]).toMatchObject(output[1]);
+  });
+});
```

Comment:
- [ ] Great Implementation, all tests are passing successfully, however, let us add few tests for remove function
![arr14](https://user-images.githubusercontent.com/55635977/131894542-f9642e39-41dd-4795-aff1-81c0b7fab2a3.png)

`For Deleting or remove function`

![arr15](https://user-images.githubusercontent.com/55635977/131894349-e9ef8d41-78b8-4231-a8c7-4ea61a3e2c46.png)

Update index after task is deleted
Update localStorage after deleting  elements
Update dom after deleting compeleted elements

Reply:
@Emmyn5600 Thanks for the Review.
According to Requirements:

> Take a look at each of these functions. **Are they pure functions? If the answer is "yes"** then writing tests for them should be straightforward. 

Our functions are Pure Functions, We have different Functions for Updating local storage, and manipulating the dom, and they are separated from Remove Functions; So writing test for Remove is straight forward and we don't need to write the test for local storage and dom manipulations as they are different units (Functions).

Reply:
Hi @ahzia

`sample for update localstorage after deleting elements`

  test('Update localStorage after deleting  elements', () => {
    clearCompletedListener(tasksList, container)(eventMock);
    const fetchedList = localStorage.getItem(LIST_STORAGE_KEY);
    expect(fetchedList).toEqual(JSON.stringify(tasksList.list));
  });